### PR TITLE
[10.0] port pricelist_per_product to 10.0

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**

### CODE REVIEW

File: pricelist_per_product/README.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+=====================
+
+This module allows you to edit pricelist items directly in the product form
+when 'Advanced pricing based on formula' is selected for the pricelist Sale Price setting.
+
+Defaults apply when the Sale Price setting is either single sale price or different prices per
+customer segment.
+
+Usage
+=====
```
Comment (LoisRForgeFlow) [inline pricelist_per_product/README.rst:16]: No usage description?

File: pricelist_per_product/README.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/135/10.0
+
+Known issues / Roadmap
```
Comment (LoisRForgeFlow) [inline pricelist_per_product/README.rst:22]: Remove empty section

File: pricelist_per_product/models/product_pricelist.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+from odoo import api, fields, models
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    product_template_count = fields.Integer(
+        compute='_compute_product_template_count',
+        string='Template with this Pricelist',
```
Comment (LoisRForgeFlow) [inline pricelist_per_product/models/product_pricelist.py:13]: s/Template/Templates

File: pricelist_per_product/models/product_pricelist.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        help='Number of Product Template with this Pricelist',
```
Comment (LoisRForgeFlow) [inline pricelist_per_product/models/product_pricelist.py:14]: s/Product Template/Product Templates

File: pricelist_per_product/models/product_pricelist.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    )
+    product_item_ids = fields.One2many(
+        comodel_name='product.pricelist.item',
+        inverse_name='pricelist_id',
+    )
+
+    @api.multi
+    def _compute_product_template_count(self):
+        """ Count the number of ``product,template`` in the pricelist """
```
Comment (LoisRForgeFlow) [inline pricelist_per_product/models/product_pricelist.py:23]: s/product,template/product.template

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
Thanks for the migration it works fine. Regarding the module itself what I see it's that if you want to use percentage pricing you still have to go to the price list form and, additionally, it is possible to change the price from the form:
image

without noticing that what is still applying is the discount:

image

File: pricelist_per_product/models/product_pricelist.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    product_template_count = fields.Integer(
+        compute='_compute_product_template_count',
+        string='Template with this Pricelist',
+        help='Number of Product Template with this Pricelist',
+    )
+    product_item_ids = fields.One2many(
+        comodel_name='product.pricelist.item',
```
Comment (lasley) [inline pricelist_per_product/models/product_pricelist.py:17]: `string="Pricelist Items"` (or whatever you want, but something 😉 )

File: pricelist_per_product/models/product_pricelist.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        inverse_name='pricelist_id',
+    )
+
+    @api.multi
+    def _compute_product_template_count(self):
+        """ Count the number of ``product,template`` in the pricelist """
+        total = self.item_ids.mapped('product_tmpl_id').ids
```
Comment (lasley) [inline pricelist_per_product/models/product_pricelist.py:24]: This expects singleton

File: pricelist_per_product/models/product_pricelist_item.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = 'product.pricelist.item'
+
+    related_sequence = fields.Integer(
+        string='Sequence',
+        related='sequence',
+        help="Allows modification of the sequence manually as "
+             "the sequence field is difficult to modify due to 'handle'."
```
Comment (lasley) [inline pricelist_per_product/models/product_pricelist_item.py:15]: Add a trailing comma

File: pricelist_per_product/models/product_pricelist_item.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+    @api.model
+    def _get_sequence_price_grid(self, vals):
+        """ This method returns a lower sequence number for the
+        ``product.pricelist.item`` associated with the product variants
+        and a higher sequence number for the ``product.pricelist.item``
+        associated with product templates. This allows us
+        to use the ``product.pricelist.item`` for a variant over the
+        one for the template in  ``sale.order.lines`` """
+        product_id = vals.get('product_id', self.product_id)
```
Comment (lasley) [inline pricelist_per_product/models/product_pricelist_item.py:26]: This is an `api.model` method, but is using `self.product_id` which assumes a singleton recordset

File: pricelist_per_product/models/product_pricelist_item.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        product_tmpl_id = vals.get('product_tmpl_id', self.product_tmpl_id)
```
Comment (lasley) [inline pricelist_per_product/models/product_pricelist_item.py:27]: Same note re mismatch api/recordset

File: pricelist_per_product/models/product_pricelist_item.py [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+        related_sequence = 15
```
Comment (lasley) [inline pricelist_per_product/models/product_pricelist_item.py:28]: IMO these ints are fairly arbitrary yet fairly import, therefore should be constants

File: pricelist_per_product/models/product_pricelist_item.py [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+        if product_id:
+            related_sequence = 5
+        elif product_tmpl_id:
+            related_sequence = 10
+        return related_sequence
+
+    @api.model
+    def create(self, vals):
+        if vals.get('product_id'):
+            vals['related_sequence'] = self._get_sequence_price_grid(vals)
```
Comment (lasley) [inline pricelist_per_product/models/product_pricelist_item.py:38]: Would ❤️  to see a test for this

File: pricelist_per_product/models/product_pricelist_item.py [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+        return super(ProductPricelistItem, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        for item in self:
+            if vals.get('product_id'):
```
Comment (lasley) [inline pricelist_per_product/models/product_pricelist_item.py:44]: Would ❤️  to see a test for this

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks @Cedric-Pigeon - comments inline

**Comment by lasley** [comment]
@Cedric-Pigeon - any news on this?

**Comment by PierrickBrun** [comment]
Do you want my help to close this PR ?

**Comment by lasley** [comment]
@PierrickBrun - it'd be great if you made the changes requested here, then submitted a PR to acsone/10.0-pricelist_per_product-cpi. Community at it's finest there IMO

**Pull request closed without merging**